### PR TITLE
Update templates and engine, pin ASP.NET package versions, add web item templates

### LINKED ISSUE #5505: Add ASP.NET "2.0" packages to offline cache when they are available
In CLI "2.0", we have removed ASP.NET packages from the offline cache because there are no ASP.NET "2.0" templates/packages/etc.

When these are available, we should add the packages used by the Web "2.0" template into the offline cache.

### PR DESCRIPTION
**Pull request by mlorbetske**
Fixes #5505

### CODE REVIEW

File: build/compile/LzmaArchive.targets [hunk 1]
```diff
@@ -79,6 +79,7 @@
         <FilesToClean Include="$(NuGetPackagesArchiveFolder)/**/*" />
 
         <NuGetPackagesArchiveDirectory Include="$(NuGetPackagesArchiveProject)/Console" />
```
Comment (livarcocc) [inline build/compile/LzmaArchive.targets]: Do we still need a separate one for console now that we have the mvc one? I mean, is the MVC a super-set?

File: test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs [hunk 2]
```diff
@@ -117,6 +112,10 @@ public void ItRestoresTheNuGetPackagesToTheNuGetCacheFolder()
                 "microsoft.visualstudio.web.browserlink",
             };
 
+            List<string> unexpectedDirectories = new List<string>()
```
Comment (livarcocc) [inline test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs]: Maybe just remove this and the assert about the unexpected directories below?

**Comment by livarcocc** [comment]
@MattGertz can you preemptively approved so that we can merge once its ready? My comments are minor.

File: build/compile/LzmaArchive.targets (hunk 1 above)
Comment (mlorbetske) [inline build/compile/LzmaArchive.targets]: I'll remove it

File: test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs (hunk 2 above)
Comment (mlorbetske) [inline test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs]: Will do

**Comment by mlorbetske** [comment]
@dotnet-bot test OpenSUSE42.1 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.10 x64 Debug Build please
@dotnet-bot test CentOS7.1 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by livarcocc** [comment]
@mlorbetske The ubuntu failure seems related to your changes:

```
Expected command to pass but it did not.
File Name: /mnt/j/workspace/dotnet_cli/master/debug_ubuntu16.10_x64_prtest/artifacts/ubuntu.16.10-x64/stage2/dotnet
Arguments: new classlib -o "/mnt/j/workspace/dotnet_cli/master/debug_ubuntu16.10_x64_prtest/test/dotnet-list-reference.Tests/bin/Debug/netcoreapp2.0/WhenNoProjectReferencesArePresentInTheProjectItPrintsError/temp" --debug:ephemeral-hive --no-restore --debug:ephemeral-hive
WorkingDir:: /mnt/j/workspace/dotnet_cli/master/debug_ubuntu16.10_x64_prtest/test/dotnet-list-reference.Tests/bin/Debug/netcoreapp2.0/WhenNoProjectReferencesArePresentInTheProjectItPrintsError/temp
Exit Code: 139
StdOut:
Getting ready...
The template "Class library" was created successfully.
```

**Comment by mlorbetske** [comment]
@livarcocc taking a look at this, it's strange, 139 isn't one of our return codes.

**Comment by mlorbetske** [comment]
@livarcocc this looks like a segfault (basing this on https://github.com/dotnet/coreclr/issues/5672), no idea where it would be coming from yet though.

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot test CentOS7.1 x64 Debug Build please
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by livarcocc** [comment]
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

Trying this again with a new runtime.

**Comment by eerhardt** [comment]
test Ubuntu16.10 x64 Debug Build

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by eerhardt** [comment]
@mlorbetske - can you try either rebasing your changes on the latest or merging master into your branch?  Just a crazy thought since I did that in #6384 and that seemed to help

**Comment by mlorbetske** [comment]
@eerhardt will do, doing a local build with the fix for the warning you saw as well

**Comment by mlorbetske** [comment]
@dotnet-bot test OpenSUSE42.1 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Pull request merged**